ETTrace/ETTrace/EMGPerfAnalysis.mm

```diff
@@ -61,7 +67,68 @@ + (void)recordStack
     sStacksLock.unlock();
 }
 
-+ (void)setupStackRecording
++ (void)recordStackForAllThreads
+{
+    thread_act_array_t threads;
+    mach_msg_type_number_t thread_count;
+    if (task_threads(mach_task_self(), &threads, &thread_count) != KERN_SUCCESS) {
+        thread_count = 0;
+    }
+    
+    // Suspend all threads but ETTrace's
+    for (mach_msg_type_number_t i = 0; i < thread_count; i++) {
+        if (threads[i] != sETTraceThread)
+            thread_suspend(threads[i]);
+    }
+    
+    CFTimeInterval time = CACurrentMediaTime();
+    for (mach_msg_type_number_t i = 0; i < thread_count; i++) {
+        if (threads[i] == sETTraceThread)
+            continue;
+        
+        Stack stack;
+        stack.time = time;
+        FIRCLSWriteThreadStack(threads[i], stack.frames, kMaxFramesPerStack, &(stack.frameCount));
+        
+        std::vector<Stack> *threadStack;
+        sStacksLock.lock();
+        if (sThreadsMap->find(threads[i]) == sThreadsMap->end()) {
+            threadStack = new std::vector<Stack>;
+            threadStack->reserve(400);
+            sThreadsMap->insert(std::pair<unsigned int, std::vector<Stack> *>(threads[i], threadStack));
+        } else {
+            threadStack = sThreadsMap->at(threads[i]);
+        }
+
+        try {
+            threadStack->emplace_back(stack);
+        } catch (const std::length_error& le) {
+            fflush(stdout);
+            fflush(stderr);
+            throw le;
+        }
+        sStacksLock.unlock();
+        
+//        // Get thread Name
```

Comment:
Can we get the thread name only the first time we sample the thread (when we reserve the 400 stack size above)?

Reply:
Oh yeah, for sure, it is just random code I used to get info if the threads.

but I wanted to keep it in the code while we decide on the final information we are going to save and the data model